src/ipc/ipc.c

```diff
@@ -315,21 +316,29 @@ static int ipc_comp_to_buffer_connect(struct ipc_comp_dev *comp,
 {
 	int ret;
 
-	if (comp->core != buffer->core) {
-		trace_ipc_error("ipc_comp_to_buffer_connect() error: cores don't match %d != %d",
-				comp->core, buffer->core);
-		return -EINVAL;
-	}
-
 	if (!cpu_is_me(comp->core))
 		return ipc_process_on_core(comp->core);
 
 	trace_ipc("ipc: comp sink %d, source %d  -> connect", buffer->id,
 		  comp->id);
 
+	if (buffer->core != comp->core) {
+		dcache_invalidate_region(buffer->cb, sizeof(*buffer->cb));
+
+		buffer->cb->inter_core = true;
+
+		if (!comp->cd->is_shared) {
+			comp->cd = comp_make_shared(comp->cd);
+			if (!comp->cd)
+				return -ENOMEM;
```

Comment:
I guess this is fatal, should we trace_error it ?

Reply:
Trace is inside of `comp_make_shared`.